dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/versioning/Navigation.java

```diff
@@ -124,8 +125,10 @@ public SourceValidity getValidity()
 		            }
 
                     DSpaceObject dso = HandleUtil.obtainHandle(objectModel);
-                    if(dso != null)
-                    {
+		            if(dso == null){
+                        dso = getItemById();
+                    }
+                    if (dso != null) {
                         validity.add(context, dso);
                     }
 
```

Comment:
The inserted `if` test here is intended to far, and makes the block a bit hard to follow.
Also, instead of two explicit `if` tests with logically opposite conditions, I think it would be easier to follow if we changed the whole block to an `if / else`
I will commit these minor changes directly rather than make new work for you :)

Reply:
Looks like mixed tabs/spaces indenting again.